crates/ruff_linter/src/rules/flake8_pytest_style/rules/parametrize.rs

```diff
@@ -797,30 +795,26 @@ fn handle_value_rows(
     }
 }
 
-pub(crate) fn parametrize(checker: &mut Checker, decorators: &[Decorator]) {
-    for decorator in decorators {
-        if is_pytest_parametrize(decorator, checker.semantic()) {
-            if let Expr::Call(ast::ExprCall {
-                arguments: Arguments { args, .. },
-                ..
-            }) = &decorator.expression
-            {
-                if checker.enabled(Rule::PytestParametrizeNamesWrongType) {
-                    if let [names, ..] = &**args {
-                        check_names(checker, decorator, names);
-                    }
-                }
-                if checker.enabled(Rule::PytestParametrizeValuesWrongType) {
-                    if let [names, values, ..] = &**args {
-                        check_values(checker, names, values);
-                    }
-                }
-                if checker.enabled(Rule::PytestDuplicateParametrizeTestCases) {
-                    if let [_, values, ..] = &**args {
-                        check_duplicates(checker, values);
-                    }
-                }
+pub(crate) fn parametrize(checker: &mut Checker, call: &ExprCall) {
+    if !is_pytest_parametrize(call, checker.semantic()) {
+        return;
+    }
+
+    if checker.enabled(Rule::PytestParametrizeNamesWrongType) {
+        if let Some(names) = call.arguments.find_argument("argnames", 0) {
+            check_names(checker, call, names);
+        }
+    }
+    if checker.enabled(Rule::PytestParametrizeValuesWrongType) {
+        if let Some(names) = call.arguments.find_argument("argnames", 0) {
+            if let Some(values) = call.arguments.find_argument("argvalues", 1) {
+                check_values(checker, names, values);
```

Comment:
You could do (and hope for the compiler to optimize a way the second call)
```suggestion
				let names = call.arguments.find_argument("argnames", 0) {
        let values = call.arguments.find_argument("argvalues", 1);
        
        if let (Some(names), Some(values)) = (names, values) {
                check_values(checker, names, values);
```